tests/test_webclient.py

```diff
@@ -72,14 +78,16 @@ def test_externalUnicodeInterference(self):
         elements of its return tuple, even when passed an URL which has
         previously been passed to L{urlparse} as a C{unicode} string.
         """
-        badInput = u'http://example.com/path'
-        goodInput = badInput.encode('ascii')
+        goodInput = u'http://example.com/path'
+        badInput = goodInput.encode('ascii')
+        if six.PY2:
+            goodInput, badInput = badInput, goodInput
```

Comment:
Yes, you are right - this is extremely confusing. I think I now understand the purpose of this test, reverted most changes and added a comment. I think the purpose was to test a case when the same url if parsed first as unicode, cached, and then parsed as bytes - but I think original test wrongly used ``urlparse(badInput)`` instead of ``self._parse(badInput)``